interfaces/dns_record/v0/README.md

```diff
@@ -0,0 +1,107 @@
+# `dns_record`
+
+## Usage
+
+This relation interface describes the expected behavior of any charm claiming to be able to interface with a DNS record Provider.
+
+## Direction
+
+The `dns_record` interface implements a provider/requirer pattern. The requirer is a charm that wishes to create a set of DNS records, and the provider is the charm managing those..
+```mermaid
+flowchart TD
+  Requirer -- dns-domains, dns-entries --> Provider
+  Provider -- dns-domains, dns-entries --> Requirer
+```
+
+## Behavior
+
+The following is the criteria that a Provider and Requirer need to adhere to be compatible with this interface.
+
+### Provider
+
+- Is expected to provide a list of dns-domains and/or a lister of dns-entries in the relation databag, each containing the domain, the status and optionally the status_description.
+
+### Requirer
+
+- Is expected to provide a list of dns-domains and/or a lister of dns-entries in the relation databag, each containing the domain it is requesting a DNS record for, the username and the password containing a juju secret to authenticate the request.
+
+## Relation Data
+
+### Provider
+
+[\[JSON Schema\]](./schemas/provider.json)
+
+Provider provides the result of the requirer request. It should be placed in the application databag.
+
+#### Example
+```json
+  "application-data": {
+    "dns-domains": [
+      {
+        "uuid": "550e8400-e29b-41d4-a716-446655440000",
+        "status": "failure",
+        "status_description": "incorrect username & password"
```

Comment:
We should adapt the example with the extended statuses we discussed yesterday

Reply:
Anything specific you have in mind?

Reply:
Well if the status_description is "incorrect username & password" I'd expect the status to be "invalid_credentials".
So my suggestion is to set a status and status_description that matches in the example